cli/cmd/generate_azure.go

```diff
@@ -0,0 +1,737 @@
+	}
+
+	// Azure command used to generate TF code for azure
+	generateAzureTfCommand = &cobra.Command{
+		Use:   "az",
+		Short: "Generate and/or execute Terraform code for Azure integration",
+		Long: `Use this command to generate Terraform code for deploying Lacework into new Azure environment.
+
+		By default, this command will function interactively, prompting for the required information to setup the new cloud account. In interactive mode, this command will:
+		
+		* Prompt for the required information to setup the integration
+		* Generate new Terraform code using the inputs
+		* Optionally, run the generated Terraform code:
+		  * If Terraform is already installed, the version will be confirmed suitable for use
+			* If Terraform is not installed, or the version installed is not suitable, a new version will be installed into a temporary location
+			* Once Terraform is detected or installed, Terraform plan will be executed
+			* The command will prompt with the outcome of the plan and allow to view more details or continue with Terraform apply
+			* If confirmed, Terraform apply will be run, completing the setup of the cloud account
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Generate TF Code
+			cli.StartProgress("Generating Azure Terraform Code...")
+
+			// Setup modifiers for NewTerraform constructor
+			mods := []azure.AzureTerraformModifier{
+				azure.WithAllSubscriptions(GenerateAzureCommandState.AllSubscriptions),
+				azure.WithManagementGroup(GenerateAzureCommandState.ManagementGroup),
+				azure.WithExistingStorageAccount(GenerateAzureCommandState.ExistingStorageAccount),
+				azure.WithStorageLocation(GenerateAzureCommandState.StorageLocation),
+				azure.WithActivityLogIntegrationName(GenerateAzureCommandState.ActivityLogIntegrationName),
+				azure.WithConfigIntegrationName(GenerateAzureCommandState.ConfigIntegrationName),
+			}
+
+			// Check if AD Creation is required, need to set values for current integration
+			if !GenerateAzureCommandState.CreateAdIntegration {
+				mods = append(mods, azure.WithAdApplicationId(GenerateAzureCommandState.AdApplicationId))
+				mods = append(mods, azure.WithAdApplicationPassword(GenerateAzureCommandState.AdApplicationPassword))
+				mods = append(mods, azure.WithAdServicePrincipalId(GenerateAzureCommandState.AdServicePrincipalId))
+			}
+
+			// Check subscriptions
+			if !GenerateAzureCommandState.AllSubscriptions {
+				if len(GenerateAzureCommandState.SubscriptionIds) > 0 {
+					mods = append(mods, azure.WithSubscriptionIds(GenerateAzureCommandState.SubscriptionIds))
+				}
+			}
+
+			// Check management groups
+			if GenerateAzureCommandState.ManagementGroup {
+				mods = append(mods, azure.WithManagementGroupId(GenerateAzureCommandState.ManagementGroupId))
+			}
+
+			// Check storage account
+			if GenerateAzureCommandState.ExistingStorageAccount {
+				mods = append(mods, azure.WithStorageAccountName(GenerateAzureCommandState.StorageAccountName))
```

Comment:
I believe the user is able to provide a storage account name when creating a new storage account also.
So we should allow the user to configure this whether they are providing existing storage account details or not.